dask/dataframe/optimize.py

```diff
@@ -108,7 +105,15 @@ def optimize_dataframe_getitem(dsk, keys):
         # Project columns and update blocks
         if column_projection:
             old = layers[k]
-            new = old.project_columns(columns)[0]
+            new = old.project_columns(columns)
+            if isinstance(new, (list, tuple)):
+                # Support project_columns tuple output, but raise FutureWarning
+                warnings.warn(
+                    f"{type(new)} return type for project_columns is deprecated. "
+                    f"This method should now return the projected Layer only.",
+                    FutureWarning,
```

Comment:
Is going through a deprecation cycle even necessary here?

Reply:
I guess it's just a question of whether `DataFrameLayer` subclasses are a public API right now. Based on the discussion we recently had with @ian-r-rose, I feel like basically the answer is no—people shouldn't be developing against HLG layers yet.

This little deprecation is simple and fine, but it's even simpler to not have to remember to fully remove it at some point. If internal Dask code is the only thing using this interface, seems fine to me.